README.md

```diff
@@ -1,583 +1,317 @@
+  github.com/my-user/my-repo/cmd/foo: my.registry/base/for/foo
+```
 
-## With `kind`
+## Naming Images
 
-Likewise, you can use `ko` with [kind](https://github.com/kubernetes-sigs/kind)
-to aid in rapid local iteration both locally and in small CI environments. To
-instruct `ko` to publish images into your `kind` cluster, the `KO_DOCKER_REPO`
-variable must be set to `kind.local`.
+`ko` provides a few different strategies for naming the image it pushes, to workaround certain registry limitations and user preferences:
 
-This would look something like:
+Given `KO_DOCKER_REPO=my.registry/repo`, by default, `ko publish ./cmd/app` will produce an image named like `my.registry/repo/app-<md5>`, which includes the MD5 hash of the full import path, to avoid collisions.
 
-```shell
-# Create a kind cluster
-kind create cluster
+- `--preserve-import-path` (`-P`) will include the entire importpath: `my.registry/repo/github.com/my-user/my-repo/cmd/app`
+- `--base` (`-B`) will omit the MD5 portion: `my.registry/repo/app`
+- `--bare` will only include the `KO_DOCKER_REPO`: `my.registry/repo`
 
-# Deploy to kind w/o registry.
-KO_DOCKER_REPO=kind.local ko apply -f config/
-```
+## Local Publishing Options
 
-If you want to create a `kind` cluster with a non default name, you can set the
-`KIND_CLUSTER_NAME` variable to the respective name (which is also supported by
-[`kind` itself](https://github.com/kubernetes-sigs/kind/releases/tag/v0.8.0)).
+`ko` is normally used to publish images to container image registries, identified by `KO_DOCKER_REPO`.
 
-Like with `minikube` above, a caveat of this approach is that it will not work
-if your container is configured with `imagePullPolicy: Always` because despite
-having the image locally, a pull is performed to ensure we have the latest
-version, it still exists, and that access hasn't been revoked. A workaround for
-this is to use `imagePullPolicy: IfNotPresent`, which should work well with `ko`
-in all contexts.
+`ko` can also publish images to a local Docker daemon, if available, by setting `KO_DOCKER_REPO=ko.local`.
 
-Note that images will not appear in the Docker daemon running `kind` as the
-cluster itself is running in a container that is running `containerd` inside.
-The images are loaded into the respective `containerd` daemon.
+`ko` can also publish images to a local [KinD](https://kind.sigs.k8s.io) cluster, if available, by setting `KO_DOCKER_REPO=kind.local`.
 
-## Configuration via `.ko.yaml`
+## Multi-Platform Images
 
-While `ko` aims to have zero configuration, there are certain scenarios where
-you will want to override `ko`'s default behavior. This is done via `.ko.yaml`.
+Because Go supports cross-compilation to other CPU architectures and operating systems, `ko` excels at producing multi-platform images. This allows you to build an image that can easily be run on other architectures such as arm64, s390x, PowerPC, etc., or other OSes like Windows.
```

Comment:
I'm not _super_ comfortable mentioning windows until we can actually confirm this works?

Reply:
Sure, removed.